egg(teamspeak): port and argument updates

TaktischerSpeck opened the pull request:
Teamspeak automatically binds the RAW, SSH and HTTP query port to the default values.
If you want to have more than one Teamspeak on the same node all Teamspeak servers will try / will bind these ports on the same default values.
(Default Teamspeak egg)
`|INFO    |Query         |   |listening for ssh query on 0.0.0.0:10022, [::]:10022`
`|INFO    |Query         |   |listening for http query on 0.0.0.0:10080, [::]:10080`

You can decide on which ports the server will bind, also which protocols it should use.
I added these variables to the egg config and Startup:
- Query protocols
- Query SSH port
- Query HTTP port

TaktischerSpeck committed:
Update egg-teamspeak3-server.json

Added variables for:
Query Protocols
Query SSH Port
Query HTTP Port

Added these variables to startup

trenutoo commented on database/Seeders/eggs/voice-servers/egg-teamspeak3-server.json:
```diff
@@ -53,7 +55,38 @@
             "user_viewable": true,
             "user_editable": false,
-            "rules": "required|integer|between:1,65535"
+            "rules": "required|integer|between:1,65535",
```
Ports below 1024 can't be used as they require root. This applies to all of the variables below. TS3 also doesn't try to bind to ports below 1024.

TaktischerSpeck replied:
Changed

trenutoo commented on database/Seeders/eggs/voice-servers/egg-teamspeak3-server.json:
```diff
@@ -53,7 +55,38 @@
+            "name": "Query Protocols",
+            "description": "Comma separated list of protocols that can be used to connect to the ServerQuery | \r\nPossible values are raw, ssh and http | \r\nE.g.: raw,ssh,http",
+            "env_variable": "QUERY_PROTOCOLS_VAR",
+            "default_value": "raw",
```
The default value in `ts3server.ini` is enabling all except HTTPS `raw,http,ssh`. I'd say disabling webinterface is not the best idea, especially when `raw` is planned to be disabled by default in the near future.

TaktischerSpeck replied:
When / where did they say `raw` will be disabled?
Last update was in 06/22, you sure they will push this soon?

I changed it to the default of the documentation, I would not remove raw in my experience a lot when not most users use this.

> Default: The default starts a raw or "classic" ServerQuery, and an encrypted ServerQuery using SSH

trenutoo replied:
In their Forum announcements when posting about new HTTP(S) webquery. Disabling by default in the near future, not removing it.

TaktischerSpeck replied:
Ah okay, but would you just remove it on default because of this?
I could imagine a bunch of people wondering.

trenutoo replied:
No, I would set the default values to match the current default, which in `ts3server.ini` by default was everything except https.

TaktischerSpeck replied:
I checked with a new TeamSpeak, default for the ini is `raw,http,ssh`
~~Without the ini the default should be `raw,ssh`~~, nvm they did not update the documentation sorry

So you would prefer  `raw,http,ssh` right?

trenutoo reviewed (changes requested):
The webquery and server query ports are optional and shouldn't affect running multiple servers at once if those ports are not assigned or exposed. They also can be changed in the `ts3server.ini` similar to the majority of options.

It doesn't hurt to make it a configurable option. SSH option by itself is useless without `query_ssh_rsa_host_key`.

TaktischerSpeck committed:
Update egg-teamspeak3-server.json

Ports below 1024 can't be used as they require root. This applies to all of the variables below. TS3 also doesn't try to bind to ports below 1024.

TaktischerSpeck committed:
Update egg-teamspeak3-server.json

Default: The default starts a raw or "classic" ServerQuery, and an encrypted ServerQuery using SSH.

TaktischerSpeck commented:
> It doesn't hurt to make it a configurable option. SSH option by itself is useless without `query_ssh_rsa_host_key`.

`query_ssh_rsa_host_key` only gives a path to the file which contains the key, if not set it will just create a file and key which you can then exchange for your key.
So yes it is useless, but the required step (or wished step for multiple TeamSpeak on one host) to use the SSH Query

TaktischerSpeck committed:
Update egg-teamspeak3-server.json

The default value in ts3server.ini is enabling all except HTTPS raw,http,ssh. I'd say disabling webinterface is not the best idea, especially when raw is planned to be disabled by default in the near future.

TaktischerSpeck commented:
> > It doesn't hurt to make it a configurable option. SSH option by itself is useless without `query_ssh_rsa_host_key`.
> 
> `query_ssh_rsa_host_key` only gives a path to the file which contains the key, if not set it will just create a file and key which you can then exchange for your key. So yes it is useless, but the required step (or wished step for multiple TeamSpeak on one host) to use the SSH Query

@Software-Noob what is about the `query_ssh_rsa_host_key`, it does only specify the location of the key, do you want this to be configurable too?

Is there something else I should change?

trenutoo reviewed (approved):
LGTM. Thanks for contributing :+1:

matthewpi committed:
Merge branch '1.0-develop' into update-teamspeak-egg
